pkg/pillar/types/dpc.go

```diff
@@ -547,6 +564,39 @@ type NetworkPortConfig struct {
 	TestResults
 }
 
+// EVE-defined port labels.
+const (
+	// AllPortsLabel references all device ports.
+	AllPortsLabel = "all"
+	// UplinkLabel references all management ports.
+	UplinkLabel = "uplink"
+	// FreeUplinkLabel references all management ports with 0 cost.
+	FreeUplinkLabel = "freeuplink"
+)
+
+// IsEveDefinedPortLabel returns true if the given port label is defined by EVE
+// and not by the user.
+func IsEveDefinedPortLabel(label string) bool {
+	switch label {
+	case AllPortsLabel, UplinkLabel, FreeUplinkLabel:
+		return true
+	}
+	return false
+}
+
+// AddEveDefinedSharedLabels adds EVE-defined shared labels that this port
```

Comment:
Do we re-evaluate this when the cost of a port changes?

Reply:
I have improved the function, now it is called `UpdateEveDefinedSharedLabels`. Though in reality adding should be enough since on config change we get new instance of DPC. Still it makes sense to make the method more robust.